tests/testthat/test-data-combined.R

```diff
@@ -0,0 +1,1072 @@
+# data needed for tests ----------------------------
+
+# will be used across all tests in this file
+
+# load the simulation
+sim <- loadTestSimulation("MinimalModel")
+simResults <- importResultsFromCSV(
+  simulation = sim,
+  filePaths = getTestDataFilePath("Stevens_2012_placebo_indiv_results.csv")
+)
+
+# import observed data (will return a list of DataSet objects)
+dataSet <- loadDataSetsFromExcel(
+  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012.xlsx"),
+  importerConfiguration = DataImporterConfiguration$new(getTestDataFilePath("ImporterConfiguration.xml"))
+)
+
+# same as dataSet, but with deliberately changed entries for testing
+dataSet2 <- loadDataSetsFromExcel(
+  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012v2.xlsx"),
+  importerConfiguration = DataImporterConfiguration$new(getTestDataFilePath("ImporterConfiguration.xml"))
+)
+
+# empty initialization ----------------------------
+
+test_that("dataCombined - initialization", {
+  # check class generator
+  expect_true(R6::is.R6Class(DataCombined))
+
+  # initialize empty object
+  myCombDat <- DataCombined$new()
+
+  # the active bindings should all be NULL at this stage
+  expect_null(myCombDat$groupMap)
+  expect_null(myCombDat$names)
+  expect_null(myCombDat$toDataFrame())
+  expect_output(print(myCombDat), "DataCombined:")
+
+  # can't use active bindings like this
+  expect_error(myCombDat$groupMap("x"))
+  expect_error(myCombDat$names("x"))
+  expect_error(myCombDat$dataTransformations("x"))
+
+  # can't enter a list of `SimulationResults` objects
+  expect_error(myCombDat$addSimulationResults(list("x", "y")))
+
+  # can enter a list, but only of `DataSet` objects
+  expect_error(myCombDat$addDataSets(list("x", "y")))
+
+  # `NULL`s not allowed
+  expect_error(myCombDat$addDataSets(list(NULL)))
+  expect_error(myCombDat$addSimulationResults(list(NULL)))
+})
+
+# either `DataSet` and `SimulationResults` provided -------------
+
+test_that("dataCombined - either dataSet or SimulationResults provided", {
+  # only DataSet input
+  myCombDat <- DataCombined$new()
+  myCombDat$addDataSets(dataSet[[1]])
+
+  # data transformations should already be setup with the defaults
+  expect_equal(
+    myCombDat$dataTransformations,
+    structure(
+      list(
+        name = "Stevens_2012_placebo.Placebo_total",
+        xOffsets = 0,
+        yOffsets = 0,
+        xScaleFactors = 1,
+        yScaleFactors = 1
+      ),
+      class = c("tbl_df", "tbl", "data.frame"),
+      row.names = c(NA, -1L)
+    )
+  )
+
+
+  # checking dataframe methods
+  df <- myCombDat$toDataFrame()
+  expect_s3_class(df, "data.frame")
```

Comment:
There has to be a method like is.data.frame or sthg like this. Why do we care that this is a s3 class?